ResearchKit/Common/ORKAnswerFormat.h

```diff
@@ -1431,6 +1441,104 @@ ORK_CLASS_AVAILABLE
+ 
+ This method is the designated initializer.
+ 
+ @param measurementSystem   The measurement system to use. See `ORKMeasurementSystem` for the
+ accepted values.
+ 
+ @return An initialized weight answer format.
+ */
+- (instancetype)initWithMeasurementSystem:(ORKMeasurementSystem)measurementSystem NS_DESIGNATED_INITIALIZER;
+
+/**
+ Returns an initialized weight answer format using the specified default/minimum/maximum values, 
+ additional precision, and measurement system.
+ 
+ @param defaultValue            The default value to display. When the value of this parameter is `nil`, the
+ picker displays 60 kg (or 133 lbs).
+ @param minimumValue            The minimum value that is accessible in the picker. If the value of this
+ parameter is `nil`, 0 kg (or 0 lbs) is the minimum.
+ @param maximumValue            The maximum value that is accessible in the picker. If the value of this
+ parameter is `nil`, 657 kg (or 1,450 lbs) is the maximum.
+ @param valueInterval           The interval at which the picker should display kilograms. The default is .5 
+ otherwise zero. If value interval is non-zero or if 'nil' the default is used.
+ @param additionalPrecision     Pass `YES` to allow additional precision; for the default, pass `NO`.
+ @param measurementSystem       The measurement system to use. See `ORKMeasurementSystem` for the
+ accepted values.
+ 
+ @return An initialized weight answer format.
+ */
+- (instancetype)initWithDefaultValue:(nullable NSNumber *)defaultValue
+                        minimumValue:(nullable NSNumber *)minimumValue
+                        maximumValue:(nullable NSNumber *)maximumValue
+                       valueInterval:(nullable NSNumber *)valueInterval
+                 additionalPrecision:(BOOL)additionalPrecision
+                   measurementSystem:(ORKMeasurementSystem)measurementSystem NS_DESIGNATED_INITIALIZER;
+
+/**
+ Indicates the measurement system used by the answer format.
+ */
+@property (readonly) ORKMeasurementSystem measurementSystem;
+
+/**
+ The value to use as the default in the specified measurement system.
+ 
+ When the value of this property is `nil`, 60 kg (or 133 lbs) is used as the default.
+ */
+@property (copy, readonly, nullable) NSNumber *defaultValue;
+
+/**
+ The minimum allowed value in the specified measurement system.
+ 
+ When the value of this property is `nil`, 0 kg (or 0 lbs) is the minimum.
+ */
+@property (copy, readonly, nullable) NSNumber *minimumValue;
+
+/**
+ The maximum allowed value in the specified measurement system.
+ 
+ When the value of this property is `nil`, 657 kg (or 1,450 lbs) is the maximum.
+ */
+@property (copy, readonly, nullable) NSNumber *maximumValue;
+
+/**
+ The interval at which the picker should display kilograms. (ignored for USC and additionalPrecision=YES)
+ 
+ The default is .5 otherwise zero. If value is non-zero or if 'nil' the default is used.
+ */
+@property (copy, readonly, nullable) NSNumber *valueInterval;
```

Comment:
If I understood correctly, it seems that it currently is only used as a boolean flag for choosing between 0.5kg and 1kg increments in metric (it is ignored in USC), and the actual numeric `valueInterval` is not used anywhere. Is that right?

Either:
- Consider converting to a `BOOL` named something like `useWholeMetricValues` (easier).
- Consider implementing proper metric value intervals (renaming to `metricValueInterval`) (Maybe it's not worth the effort, as I don't think users will find much benefit in being able to fine tune metric precision that much).
- Consider unifying `valueInterval` and `additionalPrecision` into a general `precision` `NS_ENUM` mask which accepts the following values
  1. `ORKPrecisionSimple` (1 kg increments / 1 lb increments with no ounces).
  2. `ORKPrecisionDefault` (0.5 kg increments / 1 lb increments with no ounces).
  3. `ORKPrecisionAdditional` (1 gram increments / lb and ounces).

(This is my preferred option, as it condenses all currently available options into a neat single enum, provides maximum flexibility, and keeps the defaults in line with iOS's on both measurement systems).

What do you think?

Let me know if you need help doing this and I'll be happy to do this part this week.

Reply:
You are correct.  I ended up using it only as a boolean flag.  Using enums would definitely make things much cleaner.